modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

```diff
@@ -54,40 +56,67 @@ public KafkaContainer withExternalZookeeper(String connectString) {
     }
 
     public String getBootstrapServers() {
-        return String.format("PLAINTEXT://%s:%s", proxy.getContainerIpAddress(), proxy.getFirstMappedPort());
+        if (port == -1) {
+            throw new IllegalStateException("You should start Kafka container first");
+        }
+        return String.format("PLAINTEXT://%s:%s", getContainerIpAddress(), port);
     }
 
     @Override
-    protected void doStart() {
-        String networkAlias = getNetworkAliases().get(0);
-        proxy = new SocatContainer()
-                .withNetwork(getNetwork())
-                .withTarget(KAFKA_PORT, networkAlias)
-                .withTarget(ZOOKEEPER_PORT, networkAlias);
+    @NonNull
+    public synchronized Network getNetwork() {
+        if (super.getNetwork() == null) {
+            // Backward compatibility
```

Comment:
Do we really want to add this for backward compatibility? IMO the network was an implementation detail before, not really part of the public API of the class.

Reply:
Is the timing risky here? As in:
(a) could the container come up automatically on the bridge network, but then return a completely different network to user code when this method is called?
(b) Is it possible that this code will be invoked after the container is started, meaning that `withNetwork` is not going to do the right thing...

Reply:
I've reverted the change, let's tackle it later (added a TODO)